whois-nrtm4-client/src/main/java/net/ripe/db/nrtm4/client/reader/UpdateNotificationFileReader.java

```diff
@@ -0,0 +1,69 @@
+package net.ripe.db.nrtm4.client.reader;
+
+import net.ripe.db.nrtm4.client.client.NrtmRestClient;
+import net.ripe.db.nrtm4.client.dao.Nrtm4ClientMirrorRepository;
+import net.ripe.db.nrtm4.client.client.NrtmVersionResponse;
+import net.ripe.db.nrtm4.client.dao.NrtmVersionInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class UpdateNotificationFileReader {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UpdateNotificationFileReader.class);
+
+    private final NrtmRestClient nrtmRestClient;
+
+    private final Nrtm4ClientMirrorRepository nrtm4ClientMirrorDao;
+
+
+    public UpdateNotificationFileReader(final NrtmRestClient nrtmRestClient,
+                                        final Nrtm4ClientMirrorRepository nrtm4ClientMirrorDao) {
+        this.nrtmRestClient = nrtmRestClient;
+        this.nrtm4ClientMirrorDao = nrtm4ClientMirrorDao;
+    }
+
+    public void readFile(){
+        final Map<String, NrtmVersionResponse> notificationFilePerSource =
+                nrtmRestClient.getNrtmAvailableSources()
+                .stream()
+                .collect(Collectors.toMap(
+                        string -> string,
+                        nrtmRestClient::getNotificationFile
+                ));
+        LOGGER.info("Succeeded to read notification files from {}", notificationFilePerSource.keySet());
+        final List<NrtmVersionInfo> nrtmLastVersionInfoPerSource = nrtm4ClientMirrorDao.getNrtmLastVersionInfo();
+
+        //TODO: [MH] Review integrity of the data checking the signature using the public key
+        notificationFilePerSource.forEach((source, updateNotificationFile) -> {
+            final NrtmVersionInfo nrtmLastVersionInfo = nrtmLastVersionInfoPerSource
+                    .stream()
+                    .filter(nrtmVersionInfo -> nrtmVersionInfo.source() != null && nrtmVersionInfo.source().equals(source))
+                    .findFirst()
+                    .orElse(null);
+
+            if (nrtmLastVersionInfo != null && !nrtmLastVersionInfo.sessionID().equals(updateNotificationFile.getSessionID())){
+                LOGGER.info("Different session");
+                nrtm4ClientMirrorDao.truncateTables();
+                return;
+            }
+
+            if (nrtmLastVersionInfo != null && nrtmLastVersionInfo.version() >= updateNotificationFile.getVersion()){
+                LOGGER.info("There is no new version associated with the source {}", source);
```

Comment:
This ca never happen, we should re-initialize as well in that case after logging

Reply:
should we start from the beginning when the version is the same as well?

Reply:
I created two, once when databaseUpdateNotificationVersion  > UNFVersion
and another when there are the same...if there are the same we just skip it

Reply:
No when version is same means no changes so return from there logging no updates happened since last run

Reply:
yes, I know that. I mean that in that situation we shouldn't re-initialize.
We have two situation here. When the version is the same and when somehow the version in database is ahead the unf